tests/end_to_end/helpers.h

```diff
@@ -240,13 +241,24 @@ typedef class xdp_md_helper : public xdp_md_t
 } xdp_md_helper_t;
 
 static int
-_test_xdp_adjust_head(xdp_md_t* ctx, int delta)
+_test_xdp_adjust_head(_In_ xdp_md_t* ctx, int delta)
 {
     ((xdp_md_helper_t*)ctx)->adjust_head(delta);
     return 0;
 }
 
-static const void* _test_ebpf_xdp_helper_functions[] = {(void*)&_test_xdp_adjust_head};
+static int
+_test_csum_diff(
+    _In_reads_bytes_opt_(from_size) const void* from,
+    int from_size,
+    _In_reads_bytes_opt_(to_size) const void* to,
+    int to_size,
+    int seed)
+{
+    return _net_ebpf_ext_csum_diff(from, from_size, to, to_size, seed);
+}
```

Comment:
function bodies should be in .c files not .h files.  The only exception is ones marked "inline", but these cannot be inline because you need the pointers in line 261

Reply:
The best I could do is pack these functions as static methods of a helper class. The _program_info_provider ctor invokes ebpf_load_provider for which the helper function table has to be populated. Since this helper is included by several test apps, moving the functions to a end_to_end.cpp will cause linker errors.